graalpython/com.oracle.graal.python/src/com/oracle/graal/python/builtins/modules/PythonCextBuiltins.java

```diff
@@ -2436,7 +2438,8 @@ static Object doDirect(VirtualFrame frame, @SuppressWarnings("unused") PythonMod
     @Builtin(name = "make_may_raise_wrapper", minNumOfPositionalArgs = 1, maxNumOfPositionalArgs = 2)
     @GenerateNodeFactory
     abstract static class MakeMayRaiseWrapperNode extends PythonBuiltinNode {
-        private static final WeakHashMap<RootCallTarget, WeakReference<RootCallTarget>> weakCallTargetMap = new WeakHashMap<>();
+        private static final Map<RootCallTarget, WeakReference<RootCallTarget>> weakCallTargetMap =
+                Collections.synchronizedMap(new WeakHashMap<RootCallTarget, WeakReference<RootCallTarget>>());
```

Comment:
All access to the map is in a `synchronized` block (see the method below), so I don't see how the concurrent access could occur. I was adding the synchronization very recently, is it possible that you were running an older version? Anyway, this node is currently being refactored by @fangerer, please don't spend time on it, the map will likely go away entirely.

Reply:
Ah. I tested against 21.0.0.2. So if you added it since then, it may no longer be relevant.